### ISSUE TITLE
any_errors_fatal doesn't work when using roles

### ISSUE THREAD

**Issue by flaviogrossi** [ansible/ansible#14100]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 2.1.0 (devel 2b02e8e2b4) last updated 2016/01/25 09:52:17 (GMT +200)
##### Ansible Configuration:

Default configuration
##### Environment:

Fedora 23
##### Summary:

any_errors_fatal doesn't stop task execution in some cases.
##### Steps To Reproduce:

Create the following two playbooks:
The first one has a single file named `working.yml`:

```
- hosts: test
  strategy: linear
  any_errors_fatal: True

  tasks:
    - fail: msg="fail on first host"
      when: hostname == "host1"

    - debug: msg="on both nodes"

    - fail: msg="fail on second host"
      when: hostname == "host2"
```

The second one uses the dir structure described as best practice:
File `notworking.yml`:

```
- hosts: test
  strategy: linear
  any_errors_fatal: True

  roles:
      - myrole
```

where the `myrole` role has the following content into a file `main.yml`

```
- fail: msg="fail on first host"
  when: hostname == "host1"

- debug: msg="on both nodes"

- fail: msg="fail on second host"
  when: hostname == "host2"
```

The accompanying hosts file is:

```
[test]
192.168.1.80 hostname=host1
192.168.1.81 hostname=host2
```
##### Expected Results:

The two playbooks both stops after the first task.
##### Actual Results:

Playbook `working.yml` stops after the first task, the other one executes the remaining tasks on the non failing host

**Comment by jimi-c** [comment]
# Closing This Ticket

Hi!

We believe the above commit should resolve this problem for you. This will also be included in the next release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by elouanKeryell-Even** [comment]
Still witnessing the problem in 2.3.

```
[root@localhost test_ansible_any_errors_fatal]# ansible --version
ansible 2.3.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Nov  6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]
```

---

```
[root@localhost test_ansible_any_errors_fatal]# ansible-playbook notworking.yml -i inventory.txt -v
Using /etc/ansible/ansible.cfg as config file

PLAY [test] *********************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************
ok: [192.168.1.92]
ok: [192.168.1.89]

TASK [myrole : fail] ************************************************************************************************************************************************
skipping: [192.168.1.89] => {"changed": false, "skip_reason": "Conditional result was False", "skipped": true}
fatal: [192.168.1.92]: FAILED! => {"changed": false, "failed": true, "msg": "fail on first host"}

TASK [myrole : debug] ***********************************************************************************************************************************************
ok: [192.168.1.89] => {
    "changed": false,
    "msg": "on both nodes"
}

TASK [myrole : fail] ************************************************************************************************************************************************
fatal: [192.168.1.89]: FAILED! => {"changed": false, "failed": true, "msg": "fail on second host"}
        to retry, use: --limit @/root/ekeryell/test_ansible_any_errors_fatal/notworking.retry

PLAY RECAP **********************************************************************************************************************************************************
192.168.1.89               : ok=2    changed=0    unreachable=0    failed=1
192.168.1.92               : ok=1    changed=0    unreachable=0    failed=1
```

**Comment by alangibson** [comment]
It looks like this might be fixed in 2.4.0: https://github.com/ansible/ansible/issues/22924

**Comment by jimi-c** [comment]
@alangibson thanks for the follow-up. If anyone else still sees this as an issue in devel/2.4, please open a new issue for this, as we don't typically reopen issues on the basis that the root cause is (most likely) different.

Thanks (and sorry for the late follow-up)!
